hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsRestOperation.java

```diff
@@ -221,17 +226,27 @@ private void completeExecute() throws AzureBlobFileSystemException {
     LOG.trace("{} REST operation complete", operationType);
   }
 
+  private void updateClientRequestHeader(AbfsHttpOperation httpOperation,
+      TracingContext tracingContext) {
+    tracingContext.generateClientRequestId();
+    httpOperation.getConnection()
+        .setRequestProperty(HttpHeaderConfigurations.X_MS_CLIENT_REQUEST_ID,
+            tracingContext.constructHeader());
```

Comment:
- constructHeader() may be called simply to display IDs, in which case it should not regenerate new clientReqId each time
- callee setting it would mean adding code to all methods to call this, and retry case would not be handled
Can be done internally within TC: have a generateClientRequestId function to create guid, and call it in constructor as well as the setRetryCount which is invoked per retry. However, had avoided this since we are populating ID variables only as and when they pass through the corresponding ABFS layers.